events/event_queue.go

```diff
@@ -169,6 +162,26 @@ func newEventQueueConsumerFunc(watchEvents []string, processBatchFunc ProcessBat
 			}
 		}
 
+		// Process sync consumers
+		if len(syncConsumers) != 0 {
+			failedEventsIDs := set.New[uuid.UUID]()
```

Comment:
This doesn't make sense for sync consumers? - I think its better to have 2 different functions one for sync and 1 for async